### ISSUE TITLE
getUserInfo() returns an XMLHttpRequest error on Flutter Web

### ISSUE THREAD

**Issue by evanholt1** [appsup-dart/openid_client#73]
I am working on a Flutter Web application, and the application is integrated with Keycloak. 

I am trying to follow the steps to authenticate the application with keycloak, using the following code:
```
import 'package:openid_client/openid_client_browser.dart';


  Future<void> authenticate() async {
    var uri = Uri.parse(Environment.keycloakUri);
    var clientId = Environment.clientId;
    var scopes = List<String>.of(['openid']);

    var issuer = await Issuer.discover(uri);
    var client = Client(issuer, clientId);

    urlLauncher(String url) async {
      if (await canLaunch(url)) {
        await launch(url, forceWebView: true);
      } else {
        throw 'Could not launch $url';
      }
    }

    var authenticator = Authenticator(client, scopes: scopes);

    var c = await authenticator.credential;

    if (c == null) {
      authenticator.authorize(); // this will redirect the browser
    } else {
      final userData = await c.getUserInfo();
    }
  }
```
but when calling getUserInfo() to get information like the user's preffered name, the response from flutter is always an `XMLHttpRequest Error`. 

Following the code i found that the function calls the proper keycloak userinfo url; `protocol/openid-connect/userinfo`, but that it does not add the token header to the authorization, resulting in the rejection of the request by keycloak.

How can I fix this issue?

**Comment by antonbashir** [comment]
Same issue

**Comment by rbellens** [comment]
Hi @evanholt1,

I am not able to reproduce this issue. I did add some examples in the examples folder which use keycloak. You could check those.

**Comment by hubertvolz** [comment]
Indeed, with your provided keycloak example everything works fine. 

But I got with my realm settings indeed the `XMLHttpRequest Error` too.

**Solution**: I had no entry in "WebOrigins" for my client as you have ("*").

As soon as I set this value in my client configuration, I got rid of the `XMLHttpRequest Error`. And, as soon as I removed this entry in your setup, I got the error as well.

I did this test with keycloak 18.0.2, 19.0.3, 20.0.2-1 and latest version. 

``
Error: XMLHttpRequest error.
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 299:10  createErrorWithStack
dart-sdk/lib/_internal/js_dev_runtime/patch/core_patch.dart 341:28            _throw
dart-sdk/lib/core/errors.dart 116:5                                           throwWithStackTrace
dart-sdk/lib/async/zone.dart 1378:11                                          callback
dart-sdk/lib/async/schedule_microtask.dart 40:11                              _microtaskLoop
dart-sdk/lib/async/schedule_microtask.dart 49:5                               _startMicrotaskLoop
dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 166:15           <fn>
``

Thanks @rbellens. Your extended example folder helped a lot.

**Comment by rbellens** [comment]
Setting `WebOrigins` to "*", might be insecure for production apps. Better to use the url of where the app is deployed.

**Comment by evanholt1** [comment]
Very Interesting!

Will try setting WebOrigins ASAP.

**Comment by evanholt1** [comment]
After Testing,

I can confirm that setting WebOrigins to a value value for the client,

does fix the XMLHttpRequest.

Thanks a lot for the help.
